plugin/api.go

```diff
@@ -612,6 +612,14 @@ type API interface {
 	// Minimum server version: 5.16
 	KVCompareAndDelete(key string, oldValue []byte) (bool, *model.AppError)
 
+	// KVSetWithOptions stores a key-value pair, unique per plugin, according to the given options.
+	// Returns (false, err) if DB error occurred
+	// Returns (false, nil) if the value was not set
+	// Returns (true, nil) if he value was set
+	//
+	// Minimum server version: 5.18
+	KVSetWithOptions(key string, newValue interface{}, options model.PluginKVSetOptions) (bool, *model.AppError)
```

Comment:
sad to see us breaking away from the consistency of using `value []byte` as being used in other `KV*` apis. But I don't see any other options.

Reply:
What is the downside of using `inferface{}` here?

Reply:
What happens if `interface{}` is not a `[]byte` and `EncodeJSON` is false?

Reply:
Then we throw an error. Added a note about this in the docs.